src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -371,7 +395,71 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 			case ITermOscPt.SetMark: {
 				this._createOrGetCommandDetection(this._terminal).handleGenericCommand({ genericMarkProperties: { disableCommandStorage: true } });
 			}
+			default: {
+				// Checking for known `<key>=<value>` pairs.
+				const { key, value } = this._parseKeyValueAssignment(command);
+
+				if (value === undefined) {
+					// No '=' was found, so it's not a property assignment.
+					return true;
+				}
+
+				switch (key) {
+					case ITermOscPt.CurrentDir:
+						// Encountered: `OSC 1337 ; CurrentDir=<Cwd> ST`
+						this._updateCwd(value);
+						return true;
+				}
+			}
 		}
+
+		// Unrecognized sequence
+		return false;
+	}
+
+	private _doHandleCmderSequence(data: string): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+
+		const [command, ...args] = data.split(';');
+		switch (command) {
+			case CmderOscPt.Code9:
+				if (args.length) {
+					this._updateCwd(args[0]);
+				}
+				return true;
+		}
+
+		// Unrecognized sequence
+		return false;
+	}
+
+	/**
+	 * Some escape sequences are not well known/documented, so we handle them in
+	 * an ad-hoc/one-off manner. Also, note that in some cases the `ident` value
+	 * itself acts like a command (not an terminal/emulator identifier). So, it
+	 * seems logical to have a slack method that deals with such cases.
+	 */
+	private _doHandleUnknownSequence(ident: number, data: string): boolean {
```

Comment:
For 7 and 9 should we have 2 functions named `_doHandleSetCwd` and `_doHandleSetWindowsFriendlyCwd`?